### PR TITLE
WIP: RHMAP-18330 admin/install guide review, mega review, not to be merged

### PR DESCRIPTION
**Pull request by pmacko1**
# **Changes splintered into several pr's, this will stay opened up until the last pr is resolved. Do not remove, a lot of comments here**

### CODE REVIEW

File: docs/digger/digger-and-rhmap.asciidoc [hunk 1]
```diff
@@ -1,127 +0,0 @@
----
```
Comment (pmacko1) [inline docs/digger/digger-and-rhmap.asciidoc]: this should not be here at all - keep digger and rhmap separated

**Comment by pmacko1** [comment]
@matskiv @laurafitzgerald could you take a look pls/tag ppl

File: docs/digger/installation/digger-install-intro.asciidoc [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+== AeroGear Digger Installation
 //Please DO NOT use 'Overview' as a section heading anywhere http://stylepedia.net/#sect-Red_Hat_Technical_Publications-Writing_Style_Guide-Overall_Book_Design-Unused_Heading_Titles
-GitHub repo with Ansible job used for Digger installation https://github.com/aerogear/digger-installer[Digger Installer tool^].
+https://github.com/aerogear/digger-installer[Digger Installer^] - GitHub repo with Ansible playbook used for Digger installation.
 
-The AeroGear Digger project allows you to setup a service that has the capability to build mobile applications using OpenShift containers.
+The AeroGear Digger project allows you to setup a CI/CD server that is able to build mobile applications using OpenShift containers.
 
-It is made of following components:
+NOTE: Digger can be installed behind HTTP/HTTPS proxy that do not require authentication.
```
Comment (matskiv) [inline docs/digger/installation/digger-install-intro.asciidoc]: IMHO we don't need this note here.

**Comment by matskiv** [comment]
I've added more reviewers because of the scope of this PR :)

**Comment by laurafitzgerald** [comment]
@pmacko1 There seems to be a lot of changes here. Could you add further detail to opening pr comment and JIRA listing all the changes made.

File: docs/digger/digger-and-rhmap.asciidoc (hunk 1 above)
Comment (laurafitzgerald) [reply docs/digger/digger-and-rhmap.asciidoc]: agreed this file should be removed

File: docs/digger/admin/installing-jenkins-plugins.asciidoc [hunk 3]
```diff
@@ -1,12 +1,20 @@
 == Installing Jenkins plugins
-Digger will by default install a number of Jenkins plugins required for its proper operation. If more Jenkins plugins are required, they can be installed by including them in Ansible playbook or using the Jenkins UI.
-
 NOTE: Plugins for Jenkins can be found at https://plugins.jenkins.io
 
+Digger will by default install a number of Jenkins plugins required for its proper operation. If more Jenkins plugins are required, they can be installed by including them as a task in Ansible playbook or using the Jenkins UI.
```
Comment (laurafitzgerald) [inline docs/digger/admin/installing-jenkins-plugins.asciidoc]: could add `digger installer will .....for its proper operation in the context of digger`

Comment (laurafitzgerald) [inline docs/digger/admin/installing-jenkins-plugins.asciidoc]: Also could we swap `Jenkins UI` and `Ansible playbook` order as they are listed below

File: docs/digger/admin/installing-jenkins-plugins.asciidoc [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
+
+=== Using Jenkins UI
+Go to `Manage Jenkins -> Manage Plugins`. If the plugin is not available in the
+`Available` tab then it   can be installed by uploading an `.hpi` file in
+`Advanced -> Upload Plugin`.
+
 === Using Ansible Playbook
-The `jenkins_plugins` parameter can be overridden to include new plugins. The
-structure of a Jenkins plugin is as follows.
+//https://github.com/aerogear/digger-installer/blob/master/configure-buildfarm/defaults/main.yml[Digger Installer Ansible playbook] add this when the url is known (after renaming repos)
```
Comment (laurafitzgerald) [inline docs/digger/admin/installing-jenkins-plugins.asciidoc]: Lets point the user to where they can update `jenkins_plugins` value. 
It's in 'aerogear-digger-installer -> configure-buildfarm -> defaults -> main.yml`. We should specify that they use the syntax shown by the default ones provided

File: docs/digger/installation/nagios-digger-health-check.asciidoc [hunk 5]
```diff
@@ -1,22 +1,22 @@
-== Verify The Installation
+== Nagios Installation Health checks
 
-To monitor the status of the Digger components, the Digger installer will also install https://www.nagios.org/[Nagios^] and several Nagios checks that monitor. At the end of the installation Ansible will trigger each of the checks and output the status of these checks. To navigate to Nagios and manually verify that all checks are passing, follow the steps below.
+To monitor the status of the Digger components, the Digger installer will also install https://www.nagios.org/[Nagios^] and several Nagios checks that monitor. At the end of the installation Ansible will trigger each of the checks and output the results of these checks.
```
Comment (laurafitzgerald) [inline docs/digger/installation/nagios-digger-health-check.asciidoc]: shoul remove the end of this at end of sentence `checks that monitor`. It's duplicated

File: docs/digger/installation/run-digger-installer.asciidoc [hunk 6]
```diff
@@ -1,22 +1,22 @@
-== Run the Digger Installer
+== Running the Digger Installer
```
Comment (laurafitzgerald) [inline docs/digger/installation/run-digger-installer.asciidoc]: Let's remove `the` here. sounds a bit funny.

File: docs/digger/installation/run-digger-installer.asciidoc [hunk 7]
```diff
@@ -1,22 +1,22 @@
...
 
 .To run the Digger installer, follow these steps:
 
 . Clone the https://github.com/aerogear/digger-installer[Digger Installer GitHub repo^]
```
Comment (laurafitzgerald) [inline docs/digger/installation/run-digger-installer.asciidoc]: same here

File: docs/digger/installation/run-digger-installer.asciidoc [hunk 8]
```diff
@@ -1,22 +1,22 @@
-== Run the Digger Installer
+== Running the Digger Installer
 
 .To run the Digger installer, follow these steps:
```
Comment (laurafitzgerald) [inline docs/digger/installation/run-digger-installer.asciidoc]: same here

File: docs/digger/installation/run-digger-installer.asciidoc [hunk 9]
```diff
@@ -26,23 +26,21 @@ ansible-playbook -i <your-inventory-sample> sample-build-playbook.yml
 +
 There are three points during the installation where there are manual steps required.
```
Comment (laurafitzgerald) [inline docs/digger/installation/run-digger-installer.asciidoc]: There are now two points where the user needs to accept the licence. When initially installing AndroidSDK and then when we sync the Android SDK versions

File: docs/digger/installation/using-http-proxy.asciidoc [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+Please read the link:known_issues[Known Issues] for more details.
+
+You can install AeroGear Digger behind proxy by modifying the variables in Ansible playbook.
+
+The installer will automatically configure the following components to use the proxy (if the component requires an outbound Internet access):
+
+* The Jenkins server
+* The Android SDK pod used to download Android SDKs
+* Container/node that will run the build jobs, by adding the environment variables such as `HTTP_PROXY` and `HTTPS_PROXY` in the inventory file. 
+https://github.com/aerogear/digger-installer/blob/master/inventory-sample[Sample Ansible Inventory file].
```
Comment (laurafitzgerald) [inline docs/digger/installation/using-http-proxy.asciidoc]: not sure if we need to link this here. We talk about this in the section `Running the Digger Installer`

File: docs/digger/installation/using-http-proxy.asciidoc [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+Set these variables in an Inventory file, or via CLI.
+
+For more details, please refer to http://docs.ansible.com/ansible/playbooks_variables.html[Ansible Playbook Variables Docs].
+
+|===
+| Variable name | Description
+| proxy_url
+| This URL is pasted into `http_proxy` and `https_proxy` environment variables.
+| proxy_host
+| The host name of the proxy server. Follow URL format `<host>:<port>` e.g. `127.0.0.1:8080`
```
Comment (laurafitzgerald) [inline docs/digger/installation/using-http-proxy.asciidoc]: proxy host should not contain the port

File: docs/digger/installation/using-http-proxy.asciidoc [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+=== Ansible Variables for HTTP/HTTPS Proxy
+
+Set these variables in an Inventory file, or via CLI.
+
+For more details, please refer to http://docs.ansible.com/ansible/playbooks_variables.html[Ansible Playbook Variables Docs].
+
+|===
+| Variable name | Description
+| proxy_url
+| This URL is pasted into `http_proxy` and `https_proxy` environment variables.
```
Comment (laurafitzgerald) [inline docs/digger/installation/using-http-proxy.asciidoc]: # For proxy_url the syntax should be:
# http://<proxy-host>:<proxy-port>
# or
# http://<proxy_user>:<proxy_pass>@<proxy-host>:<proxy-port>

File: docs/digger/installation/macos-setup.asciidoc [hunk 13]
```diff
@@ -1,59 +0,0 @@
-=== Setup macOS Servers
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-setup.asciidoc]: Why are we removing this file?

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 14]
```diff
@@ -1,29 +1,28 @@
...
 
-WARNING: The provided scripts helps with creating an OSX user. Do not use these scripts without reviewing their contents as it will allow access to your machine with a default password.
-Change the default `PASSWORD` and ensure `USER_ID` is not assigned.
+WARNING: Review the script before using it. The unmodified script will allow access to your machine with a default password. Change the default `PASSWORD` and ensure `USER_ID` is not assigned.
 
-By default, this script will create a user named `jenkins` with a `UID` of `550` and add the user to the the `sudoers` file. The users `PASSWORD` variable is set by default to `Password1`.
-To enable iOS builds, this user is added to the `admin` group.
+By default, the script will create a user named `jenkins` with a `UID` of `550` and add the user to the the `sudoers` file. The users `PASSWORD` variable is set by default to `Password1`. The user is added to the `admin` group.
 
-If using the sample script (contents provided below) you should read it carefully, understand its implications and edit it as required.
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-prereqs.asciidoc]: We should keep this line in.

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 15]
```diff
@@ -107,24 +108,19 @@ dseditgroup -o edit -a ${USERNAME} -t user ${OSX_SSH_GROUP_NAME}
...
-
-====== Connecting via Ansible
+=== Ansible root Permissions
 
 A number of tasks in the `provision-osx` Ansible role used in the installer require http://docs.ansible.com/ansible/latest/become.html[privilege escalation].
-This allows us to run commands as `sudo` user
+This allows Ansible to run commands as `sudo` user.
 
-The user created in the previous steps should be provided as a host variable to Ansible via `ansible_ssh_user`
+The name of the user created in the previous steps should be provided as a variable to Ansible in `ansible_ssh_user`.
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-prereqs.asciidoc]: let say provided in the inventory file here.

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 16]
```diff
@@ -107,24 +108,19 @@ dseditgroup -o edit -a ${USERNAME} -t user ${OSX_SSH_GROUP_NAME}
...
 
-Set `ansible_sudo_pass` variable or pass the `--ask-sudo-pass` flag when running the Ansible playbook to enable the installer to work with root permissions.
-This value should match the password set when creating the user.
+Set `ansible_sudo_pass` variable or pass the `--ask-sudo-pass` flag with the user password to enable the installer to work with root permissions.
 
+=== macOS Internet Access
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-prereqs.asciidoc]: Could we categorise this as a pre-requisite?

File: docs/digger/installation/known-issues.asciidoc [hunk 17]
```diff
@@ -5,36 +5,30 @@
 
 ==== HTTP proxy with macOS Jenkins node
 
-When configuring Jenkins to work behind a HTTP proxy, there is a https://issues.jenkins-ci.org/browse/JENKINS-31464[limitation/bug] in the way slave nodes use the Git plugins. This is because the slaves inherit proxy information from the master.
+When configuring Jenkins to run behind a HTTP proxy, there is a https://issues.jenkins-ci.org/browse/JENKINS-31464[limitation/bug] in the way slave nodes use the Git plugins. This is because the slaves inherit proxy information from the master.
```
Comment (laurafitzgerald) [inline docs/digger/installation/known-issues.asciidoc]: lets make the link here open a new tab

File: docs/digger/digger-and-rhmap.asciidoc (hunk 1 above)
Comment (pmacko1) [reply docs/digger/digger-and-rhmap.asciidoc]: resolved

File: docs/digger/installation/digger-install-intro.asciidoc (hunk 2 above)
Comment (pmacko1) [reply docs/digger/installation/digger-install-intro.asciidoc]: i would like to mention it in the intro, this is not the best way to do it

Comment (pmacko1) [reply docs/digger/installation/digger-install-intro.asciidoc]: resolved

File: docs/digger/admin/installing-jenkins-plugins.asciidoc (hunk 3 above)
Comment (pmacko1) [reply docs/digger/admin/installing-jenkins-plugins.asciidoc]: makes sense.
resolved

File: docs/digger/installation/known-issues.asciidoc (hunk 17 above)
Comment (pmacko1) [reply docs/digger/installation/known-issues.asciidoc]: nice! It mildly infuriates me as well if my current webpage is suddenly replaced by linked.
resolved

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 14 above)
Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: it is.
Resolved.

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 15 above)
Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: resolved

File: docs/digger/admin/installing-jenkins-plugins.asciidoc (hunk 4 above)
Comment (pmacko1) [reply docs/digger/admin/installing-jenkins-plugins.asciidoc]: resolved

File: docs/digger/installation/macos-setup.asciidoc (hunk 13 above)
Comment (pmacko1) [reply docs/digger/installation/macos-setup.asciidoc]: this should not have been removed... great catch.

File: docs/digger/admin/installing-jenkins-plugins.asciidoc (hunk 3 above)
Comment (laurafitzgerald) [reply docs/digger/admin/installing-jenkins-plugins.asciidoc]: The sentence is currently `Digger will by default install a number of Jenkins plugins required for it's proper operation`. Change this to `Digger Installer will by default by default install a number of Jenkins plugins required for it's proper operation in digger`

File: docs/digger/installation/run-digger-installer.asciidoc (hunk 6 above)
Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: resolved

File: docs/digger/installation/run-digger-installer.asciidoc (hunk 7 above)
Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: resolved

File: docs/digger/installation/run-digger-installer.asciidoc [hunk 18]
```diff
@@ -1,22 +1,22 @@
...
+NOTE: For a full list of variables that can be configured via the inventory file refer to link:#installer-variables[Installer Variables].
 
-. Run the playbook using the updated `inventory-sample` and follow the instructions printed in the terminal.:
+. Run the playbook using the updated `inventory-sample` and follow the instructions printed in the terminal:
 
-** Command to install digger without a macOS server
+** Command to install Digger without a macOS server
 +
 ----
 ansible-playbook -i <your-inventory-sample> sample-build-playbook.yml --skip-tags=osx-provision
```
Comment (pmacko1) [inline docs/digger/installation/run-digger-installer.asciidoc]: @laurafitzgerald @matskiv i believe osx-provision does not exist in the code.
provision-osx should be ok. wdyt

File: docs/digger/installation/macos-setup.asciidoc (hunk 13 above)
Comment (pmacko1) [reply docs/digger/installation/macos-setup.asciidoc]: i think this has to do something with osx-provision being called off in the past... not sure tho

File: docs/digger/installation/run-digger-installer.asciidoc (hunk 18 above)
Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: resolved

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 16 above)
Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: ping @laurafitzgerald. pls mark as resolved if not important

File: docs/digger/installation/macos-setup.asciidoc (hunk 13 above)
Comment (pmacko1) [reply docs/digger/installation/macos-setup.asciidoc]: resolved

File: docs/digger/installation/using-http-proxy.asciidoc (hunk 11 above)
Comment (pmacko1) [reply docs/digger/installation/using-http-proxy.asciidoc]: resolved

File: docs/digger/installation/using-http-proxy.asciidoc (hunk 12 above)
Comment (pmacko1) [reply docs/digger/installation/using-http-proxy.asciidoc]: resolved

File: docs/digger/installation/using-http-proxy.asciidoc (hunk 10 above)
Comment (pmacko1) [reply docs/digger/installation/using-http-proxy.asciidoc]: resolved

File: docs/digger/installation/using-http-proxy.asciidoc (hunk 12 above)
Comment (pmacko1) [reply docs/digger/installation/using-http-proxy.asciidoc]: only `http...` or `https` as well? @laurafitzgerald

Comment (laurafitzgerald) [reply docs/digger/installation/using-http-proxy.asciidoc]: @pmacko1 only http

File: docs/digger/admin/installing-jenkins-plugins.asciidoc (hunk 3 above)
Comment (pmacko1) [reply docs/digger/admin/installing-jenkins-plugins.asciidoc]: resolved

File: docs/digger/installation/run-digger-installer.asciidoc (hunk 9 above)
Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: `. Carry out manual steps during the installation
+
There are two points in the installation where you have to manually accept Android SDK licence.
+
.. Initial Android SDK installation
.. Sync of Android SDK versions
+`

Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: resolved

File: docs/digger/installation/using-http-proxy.asciidoc (hunk 12 above)
Comment (pmacko1) [reply docs/digger/installation/using-http-proxy.asciidoc]: resolved

File: docs/digger/installation/nagios-digger-health-check.asciidoc (hunk 5 above)
Comment (pmacko1) [reply docs/digger/installation/nagios-digger-health-check.asciidoc]: resolved

File: docs/digger/installation/run-digger-installer.asciidoc (hunk 8 above)
Comment (pmacko1) [reply docs/digger/installation/run-digger-installer.asciidoc]: resolved

**Comment by pmacko1** [comment]
installer-variables, shouldn't `general` be `buildfarm` ? @matskiv @laurafitzgerald

**Comment by pmacko1** [comment]
do we want to use red hat repos as a androidjdk source? rhel-7-server-optional-rpms

**Comment by pmacko1** [comment]
gem_packages please add description and format the default variable

**Comment by pmacko1** [comment]
buildfarm_node_mode incomplete description

**Comment by pmacko1** [comment]
proxy_device make the description easier to read

**Comment by pmacko1** [comment]
proxy_ctx - what is this, exactly?

**Comment by pmacko1** [comment]
buildfarm_path_env_var value - is correct?

**Comment by pmacko1** [comment]
todo: add "Running Installer steps individually" section

**Comment by pmacko1** [comment]
finished on "Install Ruby"

**Comment by pmacko1** [comment]
@matskiv @laurafitzgerald could you take a look pls/tag ppl

**Comment by matskiv** [comment]
@pmacko1 last batch of comments is quite hard to consume without the context :(

**Comment by laurafitzgerald** [comment]
@pmacko1 agreed with @matskiv comment above. Where did the questions come from? Are they related to the topic of the JIRA?

**Comment by matskiv** [comment]
@pmacko1 description of some variables can be  found in installer Readmes, eg.:
https://github.com/aerogear/aerogear-digger-installer/blob/8157740d6a38f284a88c6762ce560a4806341b8d/provision-osx/README.md

**Comment by pmacko1** [comment]
add reference to digger installer somewhere
https://github.com/aerogear/aerogear-digger-installer/tree/master

**Comment by pmacko1** [comment]
installer-variables changes discarded, changes present in #686

File: docs/digger/digger-and-rhmap.asciidoc (hunk 1 above)
Comment (pmacko1) [reply docs/digger/digger-and-rhmap.asciidoc]: https://github.com/aerogear/aerogear.org/pull/691

**Pull request closed without merging**